libbpf-cargo/src/gen/btf.rs

```diff
@@ -805,30 +807,22 @@ impl<'s> GenBtf<'s> {
             }
         }
 
-        writeln!(
-            def,
-            r#"#[derive(Debug, Copy, Clone, Default, PartialEq, Eq)]"#
-        )?;
-        writeln!(def, r#"#[repr({signed}{repr_size})]"#)?;
-        writeln!(
-            def,
-            r#"pub enum {name} {{"#,
-            name = self.anon_types.type_name_or_anon(&t),
-        )?;
+        writeln!(def, r#"#[derive(Debug, Copy, Clone, Default)]"#)?;
+        writeln!(def, r#"pub struct {enum_name}({signed}{repr_size});"#,)?;
 
-        for (i, value) in t.iter().enumerate() {
-            if i == 0 {
-                writeln!(def, r#"    #[default]"#)?;
```

Comment:
I think you are inadvertently changing the default to something that is not necessarily a valid constant and I believe we should preserve the existing logic instead. You will probably have to remember the name of the first constant and then generate a `Default` impl that returns it.